sqlglot/dialects/tsql.py

```diff
@@ -166,6 +170,29 @@ def _string_agg_sql(self: generator.Generator, expression: exp.GroupConcat) -> s
     return f"STRING_AGG({self.format_args(this, separator)}){order}"
 
 
+def _parse_date_delta(
+    exp_class: t.Type[E], unit_mapping: t.Optional[t.Dict[str, str]] = None
+) -> t.Callable[[t.List], E]:
+    def inner_func(args: t.List) -> E:
+        unit = seq_get(args, 0)
+        if unit and unit_mapping:
+            unit = exp.var(unit_mapping.get(unit.name.lower(), unit.name))
+
+        start_date = seq_get(args, 1)
+        if isinstance(start_date, exp.Literal) and start_date.this.isnumeric():
```

Comment:
why not start_date.is_number or start_date.is_int

also can you look at is_int and is correct? it looks off

Reply:
Good point, I like that better thanks

Reply:
Scarily, 1.5 is a valid `DATETIME` value. Not a valid `DATETIME2` value.

Reply:
Added a test for 1.5 as well, tested using sql fiddle and transpilation looks fine